### ISSUE TITLE
Clean up relative paths in example-datasets

### ISSUE THREAD

**Issue by ewallace** [riboviz/example-datasets#67]
After #27 is fixed, we need to review all the existing config.yaml files to remove all unneeded relative paths from files. Mostly this will involve deleting the string `../../riboviz/example-datasets/` that specifies relative path, or perhaps a longer string such as`../../riboviz/example-datasets/fungi/saccharomyces/`.

See also [riboviz#274](https://github.com/riboviz/riboviz/issues/274).

**Comment by FlicAnderson** [comment]
I am currently working on this in branch `latest-configs` and will be updating each config & using environment tokens to address the relative paths issue of this ticket.

**Comment by FlicAnderson** [comment]
This work involves running update_config tool and checking outputs, deleting the 2-1.yaml versions after once again checking no important details are lost, and substituting environment tokens (and adding details on what these should be set to).

Current progress through configs: 

DONE: 
- [x] simulated/mok/Mok-simYAL5_config.yaml 
- [x] simulated/mok/Mok-tinysim_config.yaml
- [x] (TO MERGE FROM MAIN BRANCH) fungi/saccharomyces/Zou_2020_Glucose-restricted_RPF_6-samples-CDS_w_250utrs_config.yaml

IN PROGRESS: 

TO DO: 
- [ ] archaea/haloferax/Gelsinger_etal_2020_FFC_MNase_samples_CDS_w_25utrs.yaml 
- [ ] bacteria/escherichia/Woolstenhulme_2015_3_prime_mapping_RPF_2-samples_CDS_w_25utrs.yaml 
- [ ] bacteria/escherichia/Morgan_etal_2018_heat_stress_9-samples_CDS_w_25utrs.yaml
- [ ] bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs.yaml 
- [ ] fungi/candida/Muzzey_2014_RPF_3-samples_CDS_with_120utrs_config.yaml
- [ ] fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml
- [ ] fungi/cryptococcus/Wallace_2020_JEC21_2-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml
- [ ] fungi/histoplasma/Sil_2015_RPF_4-samples_CDS_w_250utrs_config.yaml
- [ ] fungi/neurospora/Yu_2015_Nc_CDSplus120_config.yaml
- [ ] fungi/saccharomyces/Brar_2012_Meiosis_RPF_6-samples_CDS_w_250utrs_config.yaml
- [ ]  fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config.yaml 
- [ ]  fungi/saccharomyces/Jan_2014_sec63_RPF_1-sample_CDS_w_250utrs_config.yaml
- [ ]  fungi/saccharomyces/Lareau_2014_Replicates_RPF_3-samples_CDS_w_250utrs_config.yaml
- [ ]  fungi/saccharomyces/Weinberg_2016_RPF_1_sample_cerevisiae_CDS_w_250utrs_config.yaml
- [ ] fungi/schizosaccharomyces/Duncan_2018_amino_acid_starved_CDS_w_250utrs_config.yaml

**Comment by ewallace** [comment]
Thinking about this issue, clearing it up would provide some good examples of

1. running new data.
2. collaborating on github.
3. fixing filepaths (necessary and unavoidable)

Maybe good first issue for @B209678-2021 and/or @Maberuiz?

I think in most cases we would only need to check with the `--validate-only` flag. But some of them would also be relevant to the project and might want to be run in entirety.

I do not know how `--validate-only` fails or behaves unless data are downloaded.

**Comment by FlicAnderson** [comment]
I agree - this is probably a nice starter task, and will give practice in the useful skills @ewallace mentioned and getting to know the yaml files, as well as how to run some of riboviz' related tools (e.g. update_config one in this case).

**Comment by FlicAnderson** [comment]
How I'd go about this: 
- [ ] create new example-datasets branch from `main` with name relating to the task & '-67' on the end
- [ ] use [upgrade-config-tool](https://github.com/riboviz/riboviz/blob/develop/docs/user/upgrade-config.md) from `main` branch on each yaml file to update it. 
- [ ] fix relative paths in each yaml by using environment tokens 
- [ ] commit after EACH yaml file, with informative message (just in case). 
- [ ] (test these yamls after each one is completed to check they work)

**Comment by FlicAnderson** [comment]
I recommend making two branches from main: e.g. clean-paths-67-NAME and splitting the organism folders you'll work on each so there's less merge issues

**Comment by Maberuiz** [comment]
We split the different yaml files. 

In the branch clean-paths-67-ELEANOR the following yaml files will be modified:
 -Archea
 -Bacteria
 -Fungi/saccharomyces

In the branch clean-paths-67-MARCOS the following yaml files will be modified:
 -Animalia
 -Simulated
 -Fungi/candida
 -Fungi/cryptococcus
 -Fungi/histoplasma
 -Fungi/neurospora
 -Fungi/schizosaccharomyces

**Comment by Maberuiz** [comment]
A 3-1 version of all yaml files to be changed in branch clean-paths-67-MARCOS has been created using `python -m riboviz.tools.upgrade_config_file` on the original yaml files (not the 2-1 version). Still need to change the paths in the files for the environmental variables and check they work.

**Comment by Eldonoho99** [comment]
E.g., -validate_only option for Jan_2014_sec63_RPF_1-sample_CDS_w_250utrs_config_3-1.yaml using environment variables

export RIBOVIZ_SAMPLES=/exports/eddie/scratch/${USER}/J-Sc-2014
mkdir ${RIBOVIZ_SAMPLES}
export RIBOVIZ_DATA=/home/${USER}/riboviz/riboviz/data
export RIBOVIZ_ORGANISMS=/home/${USER}/riboviz/example-datasets
nextflow run prep_riboviz.nf \ -params-file /home/${USER}/riboviz/example-datasets/fungi/saccharomyces/Jan_2014_sec63_RPF_1-sample_CDS_w_250utrs_config_3-1.yaml work-dir /exports/eddie/scratch/${USER}/work -ansi-log false --validate_only --skip_inputs

**Comment by FlicAnderson** [comment]
I think for the new yamls, we don't want `3-1` in the file name. 
For example: `fungi/cryptococcus/Wallace_2020_JEC21_2-samples_10p_up12dwn9_CDS_120bpL_120bpR_config_3-1.yaml` should be `fungi/cryptococcus/Wallace_2020_JEC21_2-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml`. 

I notice you're both (@Maberuiz @Eldonoho99) saying `run using the -validate_only option`. This is just a quick remider: running riboviz with the `-validate-only` flag doesn't actually run the dataset through riboviz. It only checks that the inputs seem correct, according to some tests such as whether the files exist at the paths specified in the yaml etc. 

To avoid confusion, it's good to be really clear in tickets whether you've run riboviz successfully on a specific dataset (ie done a full riboviz run), or only tested it using validate-only option (doesn't actually run the riboviz 'full' workflow). 

Have either of you run the full-size datasets fully through riboviz using the new yamls? If so, if these have completed successfully, it's worth updating the `date run` and `riboviz-version: 2.0 | COMMIT ...` parts with the versions of riboviz you've run it with, so the yamls show they've been successfully tested recently. The commit part refers to the latest commit from the riboviz branch you're using to run the dataset, in case that's not clear.

**Comment by FlicAnderson** [comment]
> 


> E.g., -validate_only option for Jan_2014_sec63_RPF_1-sample_CDS_w_250utrs_config_3-1.yaml using environment variables
> 
> export RIBOVIZ_SAMPLES=/exports/eddie/scratch/${USER}/J-Sc-2014 mkdir ${RIBOVIZ_SAMPLES} export RIBOVIZ_DATA=/home/${USER}/riboviz/riboviz/data export RIBOVIZ_ORGANISMS=/home/${USER}/riboviz/example-datasets
> 
> `nextflow run prep_riboviz.nf \ -params-file /home/${USER}/riboviz/example-datasets/fungi/saccharomyces/Jan_2014_sex63_RPF_1-sample_CDS_w_250utrs_config_3-1.yaml work-dir /exports/Eddie/scratch/${USER}/work -ansi-log false --validate-only --skip_inputs`


This is a good idea, though if you do use --validate-only and/or --skip_inputs in the example code you add, remember to explain why you're using those commands and options and how they help you verify the environment tokens are set correctly and the yaml should work :)  
@Eldonoho99 I think this task would probably be a new issue within the riboviz documentation, can you create this?

**Comment by Maberuiz** [comment]
We've set up the environmental variable RIBOVIZ_ORGANISMS in a way in which it is designed to point to `../../riboviz/example-datasets`. However, just as @ewallace mentioned in earlier message https://github.com/riboviz/example-datasets/issues/67#issue-828051887 , this variable could be assigned as a longer path like `../../riboviz/example-datasets/fungi/saccharomyces/`. This will involve more changes from the user when running different datasets but the paths will be shorter in the yaml file. We were wondering which would be the best option.

**Comment by Maberuiz** [comment]
Mundodi and Muzzey datasets for _candida albicans_ have been run successfully with the new yaml files. Results for Mundodi could be better, but that should be a problem with the quality of the dataset.

**Comment by Eldonoho99** [comment]
Checked that the 3.1 yaml files worked using the validate only option and run successively through riboviz. Outputs checked. 
- [x] archaea/haloferax/Gelsinger_etal_2020_FFC_MNase_samples_CDS_w_25utrs_3-1.yaml
- [x] bacteria/escherichia/Woolstenhulme_2015_3_prime_mapping_RPF_2-samples_CDS_w_25utrs_3-1.yaml
- [x] bacteria/escherichia/Morgan_etal_2018_heat_stress_9-samples_CDS_w_25utrs_3-1.yaml
- [ ] bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs_3-1.yaml
- [x] fungi/saccharomyces/Brar_2012_Meiosis_RPF_6-samples_CDS_w_250utrs_config.yaml
- [x] fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config_3-1.yaml
- [x] fungi/saccharomyces/Jan_2014_sec63_RPF_1-sample_CDS_w_250utrs_config_3-1.yaml
- [x] fungi/saccharomyces/Lareau_2014_Replicates_RPF_3-samples_CDS_w_250utrs_config_3-1.yaml
- [x] fungi/saccharomyces/Weinberg_2016_RPF_1_sample_cerevisiae_CDS_w_250utrs_config_3-1.yaml
- [x] fungi/saccharomyces/Zou_2020_Glucose-restricted_RPF_6-samples-CDS_w_250utrs_config_3-1.yaml
- [x] /fungi/saccharomyces/Guydosh_2014_dom34_2-samples_CDS_w_250utrs_config_3-1.yaml

**Comment by Maberuiz** [comment]
Checked that the 3.1 yaml files worked using the validate only option and run successively through riboviz. Outputs checked.

- [x]  fungi/candida/Muzzey_2014_RPF_3-samples_CDS_with_120utrs_config.yaml
- [x]  fungi/candida/Mundodi_2021_RPF_37C_CDS_with_120utrs_config.yaml  POOR RESULTS
- [x]  fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml
- [x]  fungi/cryptococcus/Wallace_2020_JEC21_2-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml
- [x]  fungi/histoplasma/Sil_2015_RPF_4-samples_CDS_w_250utrs_config.yaml  POOR RESULTS
- [x]  fungi/neurospora/Yu_2015_Nc_CDSplus120_config.yaml
- [x]  fungi/schizosaccharomyces/Duncan_2018_amino_acid_starved_CDS_w_250utrs_config.yaml
- [x]  fungi/schizosaccharomyces/Rubio_2020_stress_config.yaml
- [x]  animalia/homo/Iwasaki_etal_2019_DMSO_samples_w_annotated_UTRs.yaml
- [x]  simulated/mok/Mok-simYAL5_config.yaml
- [x]  simulated/mok/Mok-tinysim_config.yaml

**Comment by Maberuiz** [comment]
Regarding the dataset with the config file: `fungi/schizosaccharomyces/Rubio_2020_stress_config_3-1.yaml`. A barcodes file exists: `example-datasets/fungi/schizosaccharomyces/Rubio_2020_stress_barcodes.tsv`, but was not used in the original yaml file: `fungi/schizosaccharomyces/Rubio_2020_stress_config.yaml`, in which the samples were introduced as normal fq files and not multiplexed ones. After reading the original issue ticket for the addition of this dataset [(issue 69)](https://github.com/riboviz/example-datasets/issues/69), I tried to run the dataset specifying the sample sheet but maintaining the fq files as normal ones. The result I obtained are the same as the ones observed in the second results that were obtained by Emma. Anyway, it is a strange case having the barcodes in non multiplexed samples, but I don0t find evidence in the paper indicating the samples were multiplexed.

**Comment by Eldonoho99** [comment]
I was having difficulty running the updated 3.1 yaml files for the multiplexed samples:

- bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs_3-1.yaml

- fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config_3-1.yaml

This was because the sample sheet is relative to the dir_in. It may be more useful to make it path relative to the current directory such as the t_rna file as this is not the most user friendly. Might be worth providing some more documentation highlighting this issue as it has been encountered before (#308).

**Comment by Maberuiz** [comment]
> Regarding the dataset with the config file: `fungi/schizosaccharomyces/Rubio_2020_stress_config_3-1.yaml`. A barcodes file exists: `example-datasets/fungi/schizosaccharomyces/Rubio_2020_stress_barcodes.tsv`, but was not used in the original yaml file: `fungi/schizosaccharomyces/Rubio_2020_stress_config.yaml`, in which the samples were introduced as normal fq files and not multiplexed ones. After reading the original issue ticket for the addition of this dataset [(issue 69)](https://github.com/riboviz/example-datasets/issues/69), I tried to run the dataset specifying the sample sheet but maintaining the fq files as normal ones. The result I obtained are the same as the ones observed in the second results that were obtained by Emma. Anyway, it is a strange case having the barcodes in non multiplexed samples, but I don0t find evidence in the paper indicating the samples were multiplexed.

After speaking with Emma, the barcodes are described in the paper but not used in ribosome profiling so these samples are not multiplexed and the barcodes file should be deleted. Now knowing that, when I run the dataset as non-multiplexed and including the barcodes file, I obtained the same results are the good ones from Emma, I think that riboviz doesn't read the sample sheet input when the samples are introduced as non-multiplexed. Therefore, I'll re-run the dataset setting sample sheet as null and check that I obtain the same results. Then, if my hypothesis is correct, I will deleter the barcodes file and commit the final yaml.

**Comment by FlicAnderson** [comment]
@Maberuiz - that's great to know, thanks to you and @Eldonoho99 for working that out with @3mma-mack and testing it!

**Comment by Maberuiz** [comment]
After running Rubio's dataset without indicating the sample sheet, the results are just the same as before, and the same that the final ones from Emma. My conclusion with this dataset is that the samples are not multiplexed but they show clearly two conformations of ribosomes as two sizes are clearly observable in the following plot:
[read_counts_by_length.pdf](https://github.com/riboviz/example-datasets/files/8956637/read_counts_by_length.pdf)
Therefore, I will now delete the barcodes file and the old yamls. With these last corrections, my branch will be finally ready to merge.

**Comment by Maberuiz** [comment]
After finishing all the tests and changes in the yaml files, my branch (`clean-paths-67-MARCOS`) was merged into Eleanor's branch (`clean-paths-67-ELEANOR`). So, when her changes have been finished we would be able to merge all the changes easily.

**Comment by Eldonoho99** [comment]
Still having some difficulty running riboviz on the 2 multiplexed samples -[ Gupta et al., (2018) ](https://github.com/riboviz/example-datasets/blob/clean-paths-67-ELEANOR/fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config_3-1.yaml) and [Favate et al., (2021)](https://github.com/riboviz/example-datasets/blob/clean-paths-67-ELEANOR/bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs_3-1.yaml). 

The first problem I had was that the sample sheet is [relative to `dir_in`](https://github.com/riboviz/riboviz/issues/308#issuecomment-1152374690). This was fixed by copying the sample sheet into my `dir_in` in the job submission. Possibly not the most user friendly solution and may need additional documentation. 

When I tried to run [ Gupta et al., (2018) ](https://github.com/riboviz/example-datasets/blob/clean-paths-67-ELEANOR/fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config_3-1.yaml) again I am getting this error:

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/91725826/175501646-72f3ce21-380a-4e65-8649-e3f41d340a30.png">

@FlicAnderson Have you ever encountered an issue like this?

**Comment by FlicAnderson** [comment]
@Eldonoho99  hmm. Interesting. 
We've only encountered one error from the Java Runtime Environment (JRE) before, but I **don't** think it's related: [riboviz/#375](https://github.com/riboviz/riboviz/issues/375). That related to memory and VMs that Java sets up, and was a problem encountered on Eddie: "Error occurred during initialization of VM Could not allocate metaspace: 1073741824 bytes". 

This seems like something different. I googled "fatal error detected by java runtime environment SIGBUS (0x7)" and got the following top hits: 
- https://stackoverflow.com/questions/6448163/a-fatal-error-has-been-detected-by-the-java-runtime-environment-sigsegv-libjvm  
- https://kb.froglogic.com/squish/java/troubleshoot/fatal-error-detected-java-runtime-environment/  
- https://confluence.atlassian.com/confkb/java-vm-dies-with-sigbus-0x7-when-temp-directory-is-full-on-linux-815584538.html  

Given that Eddie is running low on space lately on scratch, could this be due to a hardware issue?  Have you tried re-running this? Can you try to run it locally (if your laptop/machine is up to the task?)? Can you set it to run with locations on the lab group space on Eddie instead of using scratch?

**Comment by Eldonoho99** [comment]
@FlicAnderson it appears to have been a memory issue and [Gupta et al., (2018)](fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_CDS_w_250utrs_config_3-1.yaml). 

Currently it works by adding a line in the job submission script to copy the barcodes file directly into the `dir_in` like so: 

`cp $HOME/riboviz/example-datasets/fungi/saccharomyces/Gupta_2018_tRNA_Modification_Carbon_Nitrogen_Metabolism_RPF_9-samples_barcodes.tsv G-Sc_2018/input` 

Do you think this is worth adding to the multiplexing documentation @ewallace @FlicAnderson? I know its not the most user friendly solution but it's the easiest one given that the barcodes file is relative to `dir_in`.

**Comment by Eldonoho99** [comment]
Having trouble running bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs_3-1.yaml. 

I named the sample `rep1-ribo-am-fr.fq.gz` in the Yaml file because that's what the other yaml (original and 2.1) had done but I'm getting this error:

<img width="574" alt="image" src="https://user-images.githubusercontent.com/91725826/180175568-d201dda5-c81f-44b8-982c-096b2901fce4.png">

Also tried naming it SRR13362462.fastq.gz but got the same error.

**Comment by acope3** [comment]
Can you post your yaml file?

**Comment by Eldonoho99** [comment]
> Can you post your yaml file?
> […](#)

https://github.com/riboviz/example-datasets/blob/clean-paths-67-ELEANOR/bacteria/escherichia/Favate_etal_2021_ltee_translation_multiplexed_CDS_w_25utrs_3-1.yaml

Thank you :)

**Comment by FlicAnderson** [comment]
@acope3 I wondered whether it might be worth trying it with `rep1-ribo-am-fr.fq` instead of `rep1-ribo-am-fr.fq.gz`, as I think this has tripped me up before?  I think @Eldonoho99 is going to give this a try, but if you can see anything else that might not look right, that'd be good to know.

**Comment by acope3** [comment]
I can't see anything at a quick glance that is wrong with the YAML file

Below is the relevant code from `prep_riboviz.nf`. Looks like the logic statements are set up to handle `.fq` and `.fastq`, but not if the `.gz` is appended on the end. My still-not-quite-awake take is that this agrees with what @FlicAnderson said.

```
if ((! params.validate_only) || (! params.skip_inputs)) {
        // Filter 'params.multiplex_fq_files' down to those files that exist.
        for (entry in params.multiplex_fq_files) {
            multiplex_fq = file("${dir_in}/${entry}")
            if (multiplex_fq.exists()) {
                // Use file base name as key, ensuring that if file
                // has extension '.fastq.gz' or '.fq.gz' then both
                // extensions are removed from the name.
                multiplex_id = multiplex_fq.baseName
                if (multiplex_id.endsWith(".fastq")) {
                    multiplex_id = multiplex_id - '.fastq'
                } else if (multiplex_id.endsWith(".fq")) {
                    multiplex_id = multiplex_id - '.fq'
                }
                multiplex_id_fq[multiplex_id] = multiplex_fq
            } else {
                println("No such multiplexed file: $entry")
            }
        }
        if (! multiplex_id_fq) {
            exit 1, "None of the defined multiplexed files (multiplex_fq_files) exist"
        }
    }
    if (! params.containsKey('sample_sheet')) {
        exit 1, "Undefined sample sheet (sample_sheet)"
    }
```

**Comment by Eldonoho99** [comment]
> @acope3 I wondered whether it might be worth trying it with `rep1-ribo-am-fr.fq` instead of `rep1-ribo-am-fr.fq.gz`, as I think this has tripped me up before? I think @Eldonoho99 is going to give this a try, but if you can see anything else that might not look right, that'd be good to know.

@FlicAnderson @acope3 Thank you both for the suggestions. I'll give it a go!
